### PR TITLE
models.py: add support to get metadata and metrics for TestJob

### PR DESCRIPTION
**Pull request by liuyq**
so that it's possible to get the metadata information for each job, and the measurements information
for each job.

Signed-off-by: Yongqin Liu <yongqin.liu@linaro.org>

### CODE REVIEW

File: squad_client/core/models.py [hunk 1]
```diff
@@ -569,6 +569,40 @@ def watch(self):
             backend=self.backend,
             testjob_id=self.job_id,)
 
+    __testrun_obj__ = None
+
+    @property
+    def testrun_obj(self):
```
Comment (liuyq) [inline squad_client/core/models.py:575]: I use the name of testrun_obj here to distinguish with the testrun attribute, which is the url of the api.
it seems not work to remove that testrun attribute and rename this to testrun, as the api url will be used to get the various value for the TestRun object.
Please give suggestion if you think we should have another name for this.

**Comment by liuyq** [comment]
my purpose is to get the per environment base build metadata information, with the method you suggested, it's possible
```python
def get_build_environment_metadata(build, environment, jobs_to_be_checked):
    environment_metadata = TestRunMetadata()
    job_ids_to_be_checked = [ job.job_id for job in jobs_to_be_checked ]
    testruns = build.testruns(count=ALL, prefetch_metadata=True).values()
    for testrun in testruns:
        if testrun.job_id in job_ids_to_be_checked:
            update_object(environment_metadata, testrun.metadata)
    return environment_metadata

build.environment_metadata = get_build_environment_metadata(build, environment, jobs_to_be_checked)
```
with this method, this change is not that necessary I think.
one thing is that this method needs to get all the testruns first and then filter them with the jobs we want to check,

maybe I could define a method to assign the testrun obj to the testjob obj accordingly, then we could drop this change.
How do you think about that?

**Comment by liuyq** [comment]
OK, I find a way to get the TestRun object for TestJob without looping all the testruns of the build, 
```python
  testrun_id = testjob.testrun.strip('/').split('/')[-1]
  testrun_obj = TestRun().get(testrun_id)
```
we could have this PR closed I think.

**Comment by chaws** [comment]
@liuyq maybe you can do this?
```python
testruns = build.testruns(count=ALL, job_id__in=','.join(jobs_to_be_checked))
```

**Comment by liuyq** [comment]
> @liuyq maybe you can do this?
> 
> ```python
> testruns = build.testruns(count=ALL, job_id__in=','.join(jobs_to_be_checked))
> ```

thanks, this will reduce the call of api, but it's not that convenient with the current setup.
will see if I could change to use this method in the feature.

**Comment by chaws** [comment]
Ack, let me know if you need any help with it

**Pull request closed without merging**